Sources/NIOTestUtils/NIOHTTP1TestServer.swift

```diff
@@ -271,6 +271,7 @@ extension NIOHTTP1TestServer {
             case .stopped:
                 preconditionFailure("double stopped NIOHTTP1TestServer")
             }
+            self.currentClientChannel?.close(promise: nil)
```

Comment:
Should this close be chained off the server channel? Otherwise there’s a race in which the current client channel may be closed, the server channel may then accept another, and then that channel will not be closed directly. Probably we should confirm we’ve closed the firehose _first_, and then close the remaining connection.

Reply:
@Lukasa great idea, done!

Reply:
definitely easier to reason about now.